client/rpcflags/rpcflags_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcflags
+
+import (
+	"context"
+	"encoding/pem"
+	"flag"
+	"strings"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/testonly/integration"
+	"github.com/google/trillian/testonly/setup"
+	"github.com/google/trillian/util/flagsaver"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileNotSet(t *testing.T) {
+	// Set up Trillian servers
+	const numSequencers = 2
+	serverOpts := []grpc.ServerOption{}
+	clientOpts := []grpc.DialOption{grpc.WithInsecure()}
+	logEnv, err := integration.NewLogEnvWithGRPCOptions(context.Background(), numSequencers, serverOpts, clientOpts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer logEnv.Close()
+
+	dialOpts, err := NewClientDialOptionsFromFlags()
+	if err != nil {
+		t.Errorf("Got an unexpected error: %v", err)
+	}
+
+	// Check that the returned dial options can be used to connect and make
+	// requests against the Trillian services created above.
+	conn, err := grpc.Dial(logEnv.Address, dialOpts...)
+	if err != nil {
+		t.Errorf("failed to dial %v: %v", logEnv.Address, err)
+	}
+	defer conn.Close()
+
+	adminClient := trillian.NewTrillianAdminClient(conn)
+	_, err = adminClient.ListTrees(context.Background(), &trillian.ListTreesRequest{})
+	if err != nil {
+		t.Errorf("failed to request trees from the Admin Server: %v", err)
+	}
+}
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileMissing(t *testing.T) {
+	defer flagsaver.Save().Restore()
+	err := flag.Set("tls_cert_file", "/a/missing/file")
```

Comment:
This can be folded in - `if err := ...; err != nil`

Reply:
Done.